### PR TITLE
[CALCITE-3154] RelToSqlConverter generated Nulls Last and Nulls First when convert RexOver to sql using MysqlSqlDialect.

### PR DESCRIPTION
**Pull request by wenhuitang**
This pr for issue(https://issues.apache.org/jira/browse/CALCITE-3154)

### CODE REVIEW

**Comment by julianhyde** [comment]
The first word of a commit message should rarely be “fix”.

**Comment by hsyuan** [comment]
@wenhuitang According to calcite's convention, the title should just describe the problem.

**Comment by wenhuitang** [comment]
> The first word of a commit message should rarely be “fix”.

I have addressed. Thanks.

**Comment by wenhuitang** [comment]
> @wenhuitang According to calcite's convention, the title should just describe the problem.

I have changed. Thanks.

File: core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java [hunk 1]
```diff
@@ -795,6 +795,29 @@ void addOrderItem(List<SqlNode> orderByList, RelFieldCollation field) {
       orderByList.add(toSql(field));
     }
 
+
+    void addOrderItem(List<SqlNode> orderByList, RexProgram program, RexFieldCollation field) {
```
Comment (jinxing64) [inline core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java]: private ? and how about add some doc ?

File: core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java [hunk 2]
```diff
@@ -795,6 +795,29 @@ void addOrderItem(List<SqlNode> orderByList, RelFieldCollation field) {
...
+                  field.getNullDirection() == RelFieldCollation.NullDirection.FIRST;
+        nullDirectionNode = dialect.emulateNullDirection(
+                node, first, field.getDirection().isDescending());
+      }
+      if (nullDirectionNode != null) {
+        orderByList.add(nullDirectionNode);
+        switch (field.getDirection()) {
+        case DESCENDING:
+        case STRICTLY_DESCENDING:
+          node = SqlStdOperatorTable.DESC.createCall(POS, node);
```
Comment (jinxing64) [inline core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java]: From code style, I'd prefer a  `default` case here and doc 'do-nothing'

File: core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java [hunk 3]
```diff
@@ -795,6 +795,29 @@ void addOrderItem(List<SqlNode> orderByList, RelFieldCollation field) {
       orderByList.add(toSql(field));
     }
 
+
+    void addOrderItem(List<SqlNode> orderByList, RexProgram program, RexFieldCollation field) {
+      SqlNode node = toSql(program, field.left);
+      SqlNode nullDirectionNode = null;
+      if (field.getNullDirection() != RelFieldCollation.NullDirection.UNSPECIFIED) {
+        final boolean first =
+                  field.getNullDirection() == RelFieldCollation.NullDirection.FIRST;
```
Comment (jinxing64) [inline core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java]: How about refine the indentation ?

**Comment by michaelmior** [comment]
@hsyuan More accurately, the commit message should describe the *solution* to the problem. So in this case something like "Avoid generating NULLS LAST/FIRST..."

**Comment by julianhyde** [comment]
@michaelmior I have a different (possibly contraversial) opinion on this one. If a change fixes a bug, and if the bug can be concisely described in terms that make sense to the end user, then the ideal commit message is just the bug description. My gold standard is: what is the most useful thing to put in the release notes?

Since the message describes a problem, it is clear from the context that the change is a fix for the bug.

Occasionally a message is ambiguous. E.g. "Validator converts unquoted field names to upper case". Is that the problem or the solution? So I'd reword to "Validator should convert field names to upper case" or "Validator should not convert field names to upper case".

The second paragraph of the commit message can describe how you fixed the problem. If it's not obvious from the code.

**Comment by michaelmior** [comment]
@julianhyde Fair point. Forgive me for misinterpreting the guidelines. I somewhat disagree, because I think commit message should describe what was done. That said, I agree that based on the context, a description of a bug in a commit message implies that the commit fixes the bug. I would personally reverse the two paragraphs but I think either is acceptable.

**Comment by michaelmior** [comment]
(My mistake, I closed accidentally when replying.)

File: core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java (hunk 2 above)
Comment (wenhuitang) [reply core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java]: Thanks， I have already addressed.

**Comment by wenhuitang** [comment]
Hello, @julianhyde @michaelmior @michaelmior ，Do you have time to look at this PR again? Thanks a lot.
